lib/rubocop/cop/capybara/mixin/css_selector.rb

```diff
@@ -2,141 +2,107 @@
-      # @example
-      #   specific_pesudo_classes?('disabled') # => true
-      #   specific_pesudo_classes?('first-of-type') # => false
-      def specific_pesudo_classes?(pseudo_class)
-        SPECIFIC_PSEUDO_CLASSES.include?(pseudo_class)
-      end
+          selector.delete('#').gsub(selector.scan(/[^\\]([>,+~.].*)/).join, '')
+        end
 
-      # @param selector [String]
-      # @return [Boolean]
-      # @example
-      #   id?('#some-id') # => true
-      #   id?('.some-class') # => false
-      def id?(selector)
-        selector.start_with?('#')
-      end
+        # @param selector [String]
+        # @return [Boolean]
+        # @example
+        #   id?('#some-id') # => true
+        #   id?('.some-cls') # => false
+        def id?(selector)
+          selector.start_with?('#')
+        end
 
-      # @param selector [String]
-      # @return [Boolean]
-      # @example
-      #   attribute?('[attribute]') # => true
-      #   attribute?('attribute') # => false
-      def attribute?(selector)
-        selector.start_with?('[')
-      end
+        # @param selector [String]
+        # @return [Array<String>]
+        # @example
+        #   classes('#some-id') # => []
+        #   classes('.some-cls') # => ['some-cls']
+        #   classes('#some-id.some-cls') # => ['some-cls']
+        #   classes('#some-id.cls1.cls2') # => ['cls1', 'cls2']
+        def classes(selector)
+          selector.scan(/\.([\w-]*)/).flatten
+        end
 
-      # @param selector [String]
-      # @return [Array<String>]
-      # @example
-      #   attributes('a[foo-bar_baz]') # => {"foo-bar_baz=>true}
-      #   attributes('button[foo][bar]') # => {"foo"=>true, "bar"=>true}
-      #   attributes('table[foo=bar]') # => {"foo"=>"'bar'"}
-      def attributes(selector)
-        selector.scan(/\[(.*?)\]/).flatten.to_h do |attr|
-          key, value = attr.split('=')
-          [key, normalize_value(value)]
+        # @param selector [String]
+        # @return [Boolean]
+        # @example
+        #   attribute?('[attribute]') # => true
+        #   attribute?('attribute') # => false
+        def attribute?(selector)
+          selector.start_with?('[')
         end
-      end
 
-      # @param selector [String]
-      # @return [Boolean]
-      # @example
-      #   common_attributes?('a[focused]') # => true
-      #   common_attributes?('button[focused][visible]') # => true
-      #   common_attributes?('table[id=some-id]') # => true
-      #   common_attributes?('h1[invalid]') # => false
-      def common_attributes?(selector)
-        attributes(selector).keys.difference(COMMON_OPTIONS).none?
-      end
+        # @param selector [String]
+        # @return [Array<String>]
+        # @example
+        #   attributes('a[foo-bar_baz]') # => {"foo-bar_baz=>true}
+        #   attributes('button[foo][bar]') # => {"foo"=>true, "bar"=>true}
+        #   attributes('table[foo=bar]') # => {"foo"=>"'bar'"}
+        def attributes(selector)
```

Comment:
[memo] It got a little complicated, so I changed the logic while adding comments.
https://github.com/rubocop/rubocop-rspec/pull/1481#discussion_r1046986310